### ISSUE TITLE
UnnecessaryParentheses false positive when copying with a function

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#767]
I've got the following data class:

```kotlin
data class Generator @JvmOverloads constructor(
  val name: String = "",
  val children: (ResolvedDependency) -> Boolean = { true },
)
```

I'm getting `UnnecessaryParentheses` flagged with the following: `instanceOfGenerator.copy(children = { false })`.

I can't omit the `{`, `}` around false.

**Comment by schalkms** [comment]
The `()` are reported wrong here. It has nothing to do with the curly braces.

**Comment by vanniktech** [comment]
Oh right. Curlys don't matter here.

**Comment by schalkms** [comment]
I'm currently working on a fix.

**Comment by schalkms** [comment]
Closed via 2f832bd
